target/linux/qualcommax/files/arch/arm64/boot/dts/qcom/ipq8074-sxk80.dtsi

```diff
@@ -0,0 +1,527 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/*
+ * Copyright (c) 2021, Flole <flole@flole.de>
+ * Copyright (c) 2023, Andrew Smith <gul.code@outlook.com>
+ */
+
+/dts-v1/;
+
+#include "ipq8074.dtsi"
+#include "ipq8074-ess.dtsi"
+#include "ipq8074-hk-cpu.dtsi"
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/leds/common.h>
+
+/ {
+	#address-cells = <2>;
+	#size-cells = <2>;
+
+	interrupt-parent = <&intc>;
+
+	aliases {
+		serial0 = &blsp1_uart5;
+		led-boot = &led_front_blue;
+		led-failsafe = &led_front_red;
+		led-running = &led_front_green;
+		led-upgrade = &led_front_white;
+		label-mac-device = &dp2;
+	};
+
+	chosen {
+		stdout-path = "serial0:115200n8";
+		bootargs-append = " ubi.mtd=rootfs root=/dev/ubiblock0_0";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			gpios = <&tlmm 54 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&tlmm 57 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_front_blue: front-blue {
+			gpios = <&tlmm 33 GPIO_ACTIVE_LOW>;
+		};
+
+		led_front_green: front-green {
+			gpios = <&tlmm 29 GPIO_ACTIVE_LOW>;
+		};
+
+		led_front_red: front-red {
+			gpios = <&tlmm 31 GPIO_ACTIVE_LOW>;
+		};
+
+		led_front_white: front-white {
+			gpios = <&tlmm 26 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_green: power-green {
+			gpios = <&tlmm 21 GPIO_ACTIVE_LOW>;
+			default-state = "on";
+		};
+
+		led_power_red: power-red {
+			gpios = <&tlmm 22 GPIO_ACTIVE_LOW>;
+			panic-indicator;
+		};
+	};
+};
+
+&tlmm {
+	mdio_pins: mdio-pins {
+		mdc {
+			pins = "gpio68";
+			function = "mdc";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+
+		mdio {
+			pins = "gpio69";
+			function = "mdio";
+			drive-strength = <8>;
+			bias-pull-up;
+		};
+	};
+
+	leds_pins: leds_pinmux {
+		led_power_green {
+			pins = "gpio21";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_power_red {
+			pins = "gpio22";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_white {
+			pins = "gpio26";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_green {
+			pins = "gpio29";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_red {
+			pins = "gpio31";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+		led_blue {
+			pins = "gpio33";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+	};
+};
+
+&blsp1_uart5 {
+	status = "okay";
+};
+
+&blsp1_i2c2 {
+	pinctrl-0 = <&i2c_0_pins>;
+	pinctrl-names = "default";
+
+	status = "okay";
+	
+	tlc59208f@27 {
+		#address-cells = <1>;
+		#size-cells = <0>;
+		compatible = "ti,tlc59108";
+		reg = <0x27>;
+		
+		led@0 {
+			label = "rgb:led0";
```

Comment:
Function and color please instead of labels

Reply:
I went back over this and I'm unsure how to test these out to define them accurately. Without at least one LED defined in this part of the DTS, the status LEDs defined earlier (e.g. green:status) seem to blink rather than stay lit but I can't seem to directly work with these.

Did you look at something similar with the SRK60?

Looking at the vendor firmware, it looks like they didn't pull in the TLC591XX driver, instead opting to manage it via their own ledcontrol binary (via ioctls?).

Reply:
Something similar was done on the Netgear ipq40xx based orbis and this is how it was solved over there: https://github.com/openwrt/openwrt/blob/12137cb460e5ac0ef5924458c49822b776247179/target/linux/ipq40xx/files-6.6/arch/arm/boot/dts/qcom/qcom-ipq4019-orbi.dtsi#L216

Reply:
Why are they ON by default?

Reply:
I've changed these to default-off. But I still can't seem to directly control them. For example I can turn the red status LED on via the following:

`echo "1" > /sys/class/leds/red:status/brightness`

But if I try and interact with the LEDs under tlc59208f e.g. `echo "255" > /sys/class/leds/rgb:led1/brightness` nothing happens. Is there a way I can check/test these out with userland tools?

Reply:
If I am not mistaken there is a connector on the PCB that allows connecting a RGB strip (or that's on the other Netgear unit I've looked at)..... So 3 of the channels could be for that, and some others could be the "master channel" for the other LEDs.

Reply:
That could make more sense. I did look at `qcom-ipq4019-lbr20.dts` but that device only seems to have a power LED and a 'status ring', not sure if I could assume the same layout since I have no way of easily testing which colour is which on the SXR/SXS80.

The status LED on the front of the SXR/SXS with stock firmware seems to allow control of brightness as the white LED pulses when starting up.

Reply:
I can tell you that netgear loves those "master channels" where you can turn off all LEDs at the same time, or dim them or whatever. So you might need to go through all of them and see which ones are for that, and which ones are for that onboard LED strip port (assuming you have your unit open, it is above the ethernet jacks to the right of the LEDs IIRC.